pkg/stub/handler.go

```diff
@@ -258,22 +302,73 @@ func (h *Handler) GoodConditionUpdate(srcfg *v1alpha1.SamplesResource, newStatus
 	return nil
 }
 
+// copied from k8s.io/kubernetes/test/utils/
+func (h *Handler) IsRetryableAPIError(err error) bool {
+	// These errors may indicate a transient error that we can retry in tests.
+	if kerrors.IsInternalError(err) || kerrors.IsTimeout(err) || kerrors.IsServerTimeout(err) ||
+		kerrors.IsTooManyRequests(err) || utilnet.IsProbableEOF(err) || utilnet.IsConnectionReset(err) {
+		return true
+	}
+	// If the error sends the Retry-After header, we respect it as an explicit confirmation we should retry.
+	if _, shouldRetry := kerrors.SuggestsClientDelay(err); shouldRetry {
+		return true
+	}
+	return false
+}
+
 func (h *Handler) CreateDefaultResourceIfNeeded() error {
 	// coordinate with event handler processing
 	// where it will set h.sampleResource
 	// when it completely updates all imagestreams/templates/statuses
 	h.mutex.Lock()
 	defer h.mutex.Unlock()
-	if h.samplesResource == nil {
+	deleteInProgress := h.deleteInProgress
+
+	var err error
+	var srcfg *v1alpha1.SamplesResource
+	if deleteInProgress {
```

Comment:
seems like you can just pass deleteInProgress as an arg to CreateDefaultResourceIfNeeded rather than shuffling it around as a field on the handler.

Reply:
well, I take that back since you're using it in a variety of places.  something about this feels a little too loose to me though.  i'll keep thinking about it.